### PR TITLE
Fix abilities blocked by clothing being blocked by integrated armor

### LINKED ISSUE #56919: Integrated armors from mutation break other mutations
### Describe the bug

There are numerous mutations that only activate when a certain body part is completely naked. The newly added intergrated armors (chitin, bark, fur, etc) count as clothing for the purpose of this check, messing it up.
The most ridiculous example is spiders: having epicuticle and/or other chitinous pieces completely negates the dodge bonus from furred plate and imposes dexterity penalties from arachnid arms.

### Steps to reproduce

0. Get naked.
1. Debug the Sclerotin Plate mutation with all the prereqs.
2. Debug the Furred Plate mutation.
3. Observe that you receive no bonus to dodge from it because "you are wearing clothes".
4. Debug the Arachnid Arms mutation.
5. Observe you get no dexterity bonus because "you are wearing torso armor".

### Expected behavior

Natural integrated armors don't count as wearing clothing.

### Screenshots

_No response_

### Versions and configuration

Win 10, tiles.
Build 2022-04-15-0755

### Additional context

For context, the issue was introduced with integrated armors in #56329 (@MylieDaniels). I'm 99% sure this was not an intentional change.

### PR DESCRIPTION
**Pull request by actually-a-cat**
#### Summary
Bugfixes "Fixed abilities meant to be blocked by worn clothing being blocked by natural armor"

#### Purpose of change
Fixes #56919. The list of affected checks is actually more extensive:
* Dex bonus from arachnid and insect arms
* Climbing bonuses from arachnid and insect limbs
* Dodge bonuses from furred armor
* Immunity to being downed and knocked back from tentacles
* Rooting in soil
* Nutrients from leaves in sunlight
* Speed bonus from being barefoot with padded feet
* Drinking nectar with beak or proboscis
* Auto-retaliate attacks from spines, quills etc.
* Sound effects for walking barefoot

#### Describe the solution
* The core change is that `wearing_something_on` ignores integrated armor.
* Added helper `wearing_fitting_on`, which is the same, but also allows loose (`OVERSIZE`) items
* Reviewed all call sites of `wearing_something_on`, replacing with `fiting` as needed, and/or removing ad-hoc checks for `INTEGRATED` and `OVERSIZE` (they didn't work properly)
* Removed `footwear_factor` and replaced with `is_barefoot`, which only checks footsoles and ignores integrated armor. This matches how it was used. (There is a minor loss of function here, in that there was some consideration for only wearing one shoe, but that's not really a thing currently and when it is needed it can be readded better)
* Incidentally from the above, reworked ankle-worn items to only cover the ankle and removed `NOT_FOOTWEAR` as redundant

Other minor changes:
* Arachnid and insect arms would give a dexterity *penalty*, on top of not giving a bonus, when constricted. This doesn't make sense to me and feels accidental so I removed it.
* I flipped the effect of having roots and walking on dirt, from slowing you down if you have shoes to slowing you down if you don't. That I can imagine is because your roots are trying to dig in or something, the current behavior makes no sense to me.

#### Describe alternatives you've considered
I considered further-reaching changes to armor info functions, to make them aware of integrated armor and allow anything to be queried either including or disregarding it. But I've been informed that all that stuff is due for an overhaul anyway, so I went for a minimal solution that solves the bugs with as little disruption as possible.

The whole deal with `OVERSIZE` is not ideal (why does a regular backpack restrict your spider arms but a hiking backpack doesn't?) but that flag as a whole is a problem.

#### Testing
I manually tested most of the listed scenarios (with the notable exception of tentacles holding the character down against a shockwave - no idea how to try it). It seems like some automated tests would fit in here and I might add in the (not immediate) future.

#### Additional context
I will be mostly away and unable to do any development for a while so if anything is amiss here it will have to wait.

### CODE REVIEW

**Comment by actually-a-cat** [comment]
Drafting because I also need to review uses of `is_wearing_shoes` and I might not be able to get around to it for a while

**Comment by KittyTac** [comment]
Make sure to fix Thorns + Bark too.

File: src/character.cpp [hunk 1]
```diff
@@ -7835,10 +7835,9 @@ void Character::update_vitamins( const vitamin_id &vit )
 
 void Character::rooted_message() const
 {
-    bool wearing_shoes = footwear_factor() == 1.0;
     if( ( has_trait( trait_ROOTS2 ) || has_trait( trait_ROOTS3 ) || has_trait( trait_CHLOROMORPH ) ) &&
         get_map().has_flag( ter_furn_flag::TFLAG_PLOWABLE, pos() ) &&
-        !wearing_shoes ) {
+        !is_barefoot() ) {
```
Comment (fungamer2-2) [inline src/character.cpp:7840]:
```suggestion
        is_barefoot() ) {
```

**Comment by I-am-Erk** [comment]
Is this still under construction? We may need someone to take over this PR

**Comment by actually-a-cat** [comment]
The shoes-related part of this is not ideal (I should instead make `Character::is_wearing_shoes` aware of the difference between feet and foot soles and use it) but for now I think it's ok.

File: src/character.cpp [hunk 2]
```diff
@@ -9305,7 +9303,7 @@ int Character::run_cost( int base_cost, bool diag ) const
 
         if( ( has_trait( trait_ROOTS3 ) || has_trait( trait_CHLOROMORPH ) ) &&
             here.has_flag( ter_furn_flag::TFLAG_DIGGABLE, pos() ) ) {
-            movecost += 10 * footwear_factor();
+            movecost += 10 * ( is_barefoot() ? 1 : 0 );
         }
```
Comment (Night-Pryanik) [inline src/character.cpp:9307]: This whole snippet could be rewritten as
```
if( ( has_trait( trait_ROOTS3 ) || has_trait( trait_CHLOROMORPH ) ) &&
    here.has_flag( ter_furn_flag::TFLAG_DIGGABLE, pos() ) && is_barefoot() ) {
    movecost += 10;
}
```

File: src/character_attire.cpp [hunk 3]
```diff
@@ -947,13 +955,34 @@ std::list<item> outfit::get_visible_worn_items( const Character &guy ) const
...
+            return true;
+        }
+    }
+    return false;
+}
+
+bool outfit::wearing_fitting_on( const bodypart_id &bp ) const
+{
+    for( const item &i : worn ) {
+        if( i.covers( bp ) && ( !i.has_flag( flag_INTEGRATED ) && !i.has_flag( flag_OVERSIZE ) ) ) {
```
Comment (Night-Pryanik) [inline src/character_attire.cpp:968]:
```suggestion
        if( i.covers( bp ) && !i.has_flag( flag_INTEGRATED ) && !i.has_flag( flag_OVERSIZE ) ) {
```

File: src/game.cpp [hunk 4]
```diff
@@ -4480,8 +4480,7 @@ void game::knockback( std::vector<tripoint> &traj, int stun, int dam_mult )
                                  targ->get_name() );
                     }
                 } else if( u.posx() == traj_front.x && u.posy() == traj_front.y &&
-                           ( u.has_trait( trait_LEG_TENT_BRACE ) && ( !u.footwear_factor() ||
-                                   ( u.footwear_factor() == .5 && one_in( 2 ) ) ) ) ) {
+                           ( u.has_trait( trait_LEG_TENT_BRACE ) && u.is_barefoot() ) ) {
```
Comment (Night-Pryanik) [inline src/game.cpp:4483]:
```suggestion
                           u.has_trait( trait_LEG_TENT_BRACE ) && u.is_barefoot() ) {
```

File: src/character.cpp [hunk 5]
```diff
@@ -1298,7 +1298,7 @@ int Character::swim_speed() const
         ret -= hand_bonus_mult * webbing_factor * 0.5f;
     }
     if( has_flag( json_flag_WEBBED_FEET ) ) {
-        ret -= ( 1 - footwear_factor() ) * webbing_factor * 0.5f;
+        ret -= is_barefoot() ? 1 : 0 * webbing_factor * 0.5f;
```
Comment (Night-Pryanik) [inline src/character.cpp:1301]:
```
if( has_flag( json_flag_WEBBED_FEET ) && is_barefoot() ) {
    ret -= webbing_factor * 0.5f;
}
```

**Pull request merged**